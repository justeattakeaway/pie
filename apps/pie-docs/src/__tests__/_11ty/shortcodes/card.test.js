const card = require('../../../_11ty/shortcodes/card');

const item = {
    linkText: 'label',
    href: 'link.com',
};

const itemWithImage = {
    ...item,
    src: '/path/to/image',
};

const itemOpeningInANewTab = {
    ...item,
    shouldOpenInNewTab: true,
};

describe('card.js', () => {
    it.each([
        [[item]],
        [[itemWithImage]],
        [[itemOpeningInANewTab]],
        [[item, item]]
    ])('should return the expected HTML', (items) => {
        // act
        const result = card({ items });

        // assert
        expect(result).toMatchSnapshot();
    });
});
