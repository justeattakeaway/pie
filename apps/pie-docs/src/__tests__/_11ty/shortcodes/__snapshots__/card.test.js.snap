// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`card.js should return the expected HTML 1`] = `
"<article class=\\"c-card\\">
                    
                    
                    <div class=\\"c-card-labelContainer\\">
                <p class=\\"c-card-label\\" ><span>label</span></p>
                
            </div>
                </article>"
`;

exports[`card.js should return the expected HTML 2`] = `
"<article class=\\"c-card c-card--hasImage\\">
                    
                    <img class=\\"c-card-image\\" src=\\"/path/to/image\\" role=\\"presentation\\" alt=\\"\\">
                    <div class=\\"c-card-labelContainer\\">
                <p class=\\"c-card-label\\" ><span>label</span></p>
                
            </div>
                </article>"
`;

exports[`card.js should return the expected HTML 3`] = `
"<article class=\\"c-card\\">
                    
                    
                    <div class=\\"c-card-labelContainer\\">
                <a class=\\"c-card-label\\" href=\\"link.com\\"><span>label</span></a>
                <svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 16 16\\" height=\\"21\\" width=\\"21\\" fill=\\"#000\\" aria-hidden=\\"true\\" class=\\"c-pieIcon c-pieIcon--linkExternal\\"><path fill=\\"#242E30\\" d=\\"M13.25 3.564 8.21 8.63l-.928-.928 5.093-5.066H9.662V1.324h3.833a1.093 1.093 0 0 1 1.067 1.093V6.25H13.25V3.564Z\\"></path><path fill=\\"#242E30\\" d=\\"M11.302 12.831c.343-.34.537-.804.54-1.287V7.598h1.312v3.946a3.141 3.141 0 0 1-3.142 3.132h-5.46a3.141 3.141 0 0 1-3.14-3.132V6.075a3.132 3.132 0 0 1 3.14-3.132h3.79v1.312h-3.79a1.829 1.829 0 0 0-1.802 1.82v5.469a1.829 1.829 0 0 0 1.802 1.82h5.46c.484 0 .948-.192 1.29-.533Z\\"></path></svg>
            </div>
                </article>"
`;

exports[`card.js should return the expected HTML 4`] = `
"<article class=\\"c-card\\">
                    
                    
                    <div class=\\"c-card-labelContainer\\">
                <a class=\\"c-card-label\\" href=\\"link.com\\" target=\\"_blank\\"><span>label</span></a>
                <svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 16 16\\" height=\\"21\\" width=\\"21\\" fill=\\"#000\\" aria-hidden=\\"true\\" class=\\"c-pieIcon c-pieIcon--linkExternal\\"><path fill=\\"#242E30\\" d=\\"M13.25 3.564 8.21 8.63l-.928-.928 5.093-5.066H9.662V1.324h3.833a1.093 1.093 0 0 1 1.067 1.093V6.25H13.25V3.564Z\\"></path><path fill=\\"#242E30\\" d=\\"M11.302 12.831c.343-.34.537-.804.54-1.287V7.598h1.312v3.946a3.141 3.141 0 0 1-3.142 3.132h-5.46a3.141 3.141 0 0 1-3.14-3.132V6.075a3.132 3.132 0 0 1 3.14-3.132h3.79v1.312h-3.79a1.829 1.829 0 0 0-1.802 1.82v5.469a1.829 1.829 0 0 0 1.802 1.82h5.46c.484 0 .948-.192 1.29-.533Z\\"></path></svg>
            </div>
                </article>"
`;

exports[`card.js should return the expected HTML 5`] = `
"<article class=\\"c-card\\">
                    
                    
                    <div class=\\"c-card-labelContainer c-card-labelContainer--internalLink\\">
                <a class=\\"c-card-label\\" href=\\"link.com\\"><span>label</span></a>
                <svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 16 16\\" height=\\"16\\" width=\\"16\\" fill=\\"#000\\" aria-hidden=\\"true\\" class=\\"c-pieIcon c-pieIcon--arrowRight\\"><path fill=\\"#242E30\\" d=\\"M1 8.75h11.65l-3.76 3.76L10 13.57l4.68-4.69a1.239 1.239 0 0 0 0-1.76L10 2.43 8.89 3.49l3.76 3.76H1v1.5Z\\"></path></svg>
            </div>
                </article>"
`;

exports[`card.js should return the expected HTML 6`] = `
"<article class=\\"c-card\\">
                    <div class=\\"c-card-container\\">
            <svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 16 16\\" height=\\"48\\" width=\\"48\\" fill=\\"#000\\" class=\\"c-pieIcon c-pieIcon--bulbLightning c-card-icon c-card-icon--support-brand-03\\" aria-hidden=\\"true\\"><path fill=\\"#242E30\\" d=\\"M10.162 13.469H5.838l.35 1.312h3.622l.35-1.312Z\\"></path><path fill=\\"#242E30\\" d=\\"m5.375 10.117.35 2.04h4.611l.29-2.048c.07-.393.267-.752.56-1.024a4.523 4.523 0 0 0 1.408-3.273 4.594 4.594 0 0 0-9.126-.76A4.533 4.533 0 0 0 4.77 9.067c.306.278.517.645.604 1.05Zm-.612-4.856a3.281 3.281 0 0 1 6.518.551 3.239 3.239 0 0 1-.988 2.337c-.49.47-.821 1.082-.945 1.75l-.158.954H6.801l-.148-.945a3.238 3.238 0 0 0-.963-1.75 3.281 3.281 0 0 1-.927-2.897Z\\"></path></svg>
            <h2 class=\\"c-card-heading\\">heading</h2>
            <p class=\\"c-card-content\\">Content</p>
        </div>
                    
                    <div class=\\"c-card-labelContainer\\">
                <p class=\\"c-card-label\\" ><span>label</span></p>
                
            </div>
                </article>"
`;

exports[`card.js should return the expected HTML 7`] = `
"<div class=\\"c-card-wrapper\\">
                    <ul class=\\"c-card-list c-card-list--fillContainer\\">
                        <li><article class=\\"c-card\\">
                    
                    
                    <div class=\\"c-card-labelContainer\\">
                <p class=\\"c-card-label\\" ><span>label</span></p>
                
            </div>
                </article></li><li><article class=\\"c-card\\">
                    
                    
                    <div class=\\"c-card-labelContainer\\">
                <p class=\\"c-card-label\\" ><span>label</span></p>
                
            </div>
                </article></li><li><article class=\\"c-card\\">
                    
                    
                    <div class=\\"c-card-labelContainer\\">
                <p class=\\"c-card-label\\" ><span>label</span></p>
                
            </div>
                </article></li>
                    </ul>
                </div>"
`;

exports[`card.js should return the expected HTML 8`] = `
"<div class=\\"c-card-wrapper\\">
                    <ul class=\\"c-card-list\\">
                        <li><article class=\\"c-card\\">
                    
                    
                    <div class=\\"c-card-labelContainer\\">
                <p class=\\"c-card-label\\" ><span>label</span></p>
                
            </div>
                </article></li><li><article class=\\"c-card\\">
                    
                    
                    <div class=\\"c-card-labelContainer\\">
                <p class=\\"c-card-label\\" ><span>label</span></p>
                
            </div>
                </article></li>
                    </ul>
                </div>"
`;
